pkg/ibm_plugin/server/plugin_integration_test.go

```diff
@@ -0,0 +1,406 @@
+	// removes all of paraglide's deployments on IBM
+	region, err := ibmCommon.ZoneToRegion(zone)
+	require.NoError(t, err)
+
+	defer func() {
+		err := sdk.TerminateParagilderDeployments(resourceGroupID, region)
+		require.NoError(t, err)
+	}()
+
+	orchestratorServerConfig := config.Config{
+		Server: config.Server{
+			Host:    "localhost",
+			Port:    "8080",
+			RpcPort: "8081",
+		},
+		TagService: config.TagService{
+			Host: "localhost",
+			Port: strconv.Itoa(taggingPort),
+		},
+		KVStore: config.TagService{
+			Port: strconv.Itoa(kvstorePort),
+			Host: "localhost",
+		},
+		CloudPlugins: []config.CloudPlugin{
+			{
+				Name: utils.IBM,
+				Host: "localhost",
+				Port: strconv.Itoa(IBMServerPort),
+			},
+		},
+		Namespaces: map[string][]config.CloudDeployment{
+			testNamespace: {
+				{
+					Name:       utils.IBM,
+					Deployment: testDeployment,
+				},
+			},
+		},
+	}
+
+	// start controller server
+	fmt.Println("Setting up controller server and kvstore server")
+	orchestratorServerAddr := orchestratorServerConfig.Server.Host + ":" + orchestratorServerConfig.Server.RpcPort
+	orchestrator.Setup(orchestratorServerConfig, true)
+
+	// start ibm plugin server
+	fmt.Println("Setting up IBM server")
+	ibmServer := Setup(IBMServerPort, orchestratorServerAddr)
+
+	fmt.Println("Setting up kv store server")
+	tagging.Setup(dbPort, taggingPort, true)
+
+	fmt.Println("Setting up kv tagging server")
+	kvstore.Setup(dbPort, kvstorePort, true)
+
+	// Create IBM VM
+	fmt.Println("\nCreating IBM VM...")
+	imageIdentity := vpcv1.ImageIdentityByID{ID: &image}
+	zoneIdentity := vpcv1.ZoneIdentityByName{Name: &zone}
+	myTestProfile := string(testProfile)
+
+	testPrototype := &vpcv1.InstancePrototypeInstanceByImage{
+		Image:   &imageIdentity,
+		Zone:    &zoneIdentity,
+		Name:    core.StringPtr(instanceName),
+		Profile: &vpcv1.InstanceProfileIdentityByName{Name: &myTestProfile},
+	}
+
+	description, err := json.Marshal(vpcv1.CreateInstanceOptions{InstancePrototype: vpcv1.InstancePrototypeIntf(testPrototype)})
+	require.NoError(t, err)
+
+	resource := &paragliderpb.CreateResourceRequest{Name: instanceName, Deployment: &paragliderpb.ParagliderDeployment{Id: testDeployment, Namespace: testNamespace}, Description: description}
+	res, err := ibmServer.CreateResource(context.Background(), resource)
+	require.NoError(t, err)
+	require.NotNil(t, res)
+	// append instance's ID
+	URIParts := strings.Split(res.Uri, "/")
+	resID := IBMResourceIDPrefix + URIParts[len(URIParts)-1]
+
+	// Add permit list for IBM VM
+	fmt.Println("Adding IBM permit list rules...")
+
+	addRulesRequest := &paragliderpb.AddPermitListRulesRequest{
+		Namespace: testNamespace,
+		Resource:  resID,
+		Rules:     pingTestPermitList,
+	}
+
+	resp, err := ibmServer.AddPermitListRules(context.Background(), addRulesRequest)
+	require.NoError(t, err)
+	require.NotNil(t, resp)
+
+	utils.Log.Printf("Test response: %+v", resp)
+}
+
+// TODO(cohen-j-omer) will add IBM connectivity check method
```

Comment:
If IBM connectivity check is complicated to write, then we can just run an Azure connectivity check. Ping is bidirectional so this should be ok.

Reply:
Sounds good.